.circleci/config.yml

```diff
@@ -0,0 +1,74 @@
+version: 2
+jobs:
+  setup:
+    docker:
+      - image: circleci/node:8
+        environment:
+          NODE_ENV: test
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - dependency-cache-{{ checksum "package-lock.json"}}
+            - dependency-cache-
+      - run:
+          name: Install node dependencies
+          command: |
+            npm ci
+      - save_cache:
+          key: dependency-cache-{{ checksum "package-lock.json" }}
+          paths:
+            - ./node_modules
+      - persist_to_workspace:
+          root: ./
+          paths:
+            - .
+  publish:
+    docker:
+      - image: circleci/node:8
+        environment:
+          NODE_ENV: test
+    steps:
+      - attach_workspace:
+          at: ./
+      - run:
+          name: npm publish via script
+          command: |
+            ./scripts/release.sh
+  test:
+    docker:
+      - image: node:8.9.4
+        environment:
+          NODE_ENV: test
+          MOCHA_OPTS: --reporter mocha-junit-reporter --reporter-options mochaFile=$CIRCLE_TEST_REPORTS/mocha/junit.xml
+          CODECOV_TOKEN: 93320c3b-2cd0-48d7-9ff1-5b29632d377e
```

Comment:
Hopefully it's ok to have the CODECOV_TOKEN checked in?

Reply:
seems bad, this belongs as an environment variable in circleci settings rather than within the source

Reply:
let me know if you don't know how to do that and I can explain @Shockolate